core/trino-main/src/main/java/io/trino/execution/DropTableTask.java

```diff
@@ -76,17 +75,18 @@ public ListenableFuture<Void> execute(
                     "Table '%s' does not exist, but a view with that name exists. Did you mean DROP VIEW %s?", originalTableName, originalTableName);
         }
 
-        RedirectionAwareTableHandle redirectionAwareTableHandle = metadata.getRedirectionAwareTableHandle(session, originalTableName);
-        if (redirectionAwareTableHandle.getTableHandle().isEmpty()) {
+        QualifiedObjectName targetTable = metadata.getRedirectedTableName(session, originalTableName);
+        boolean tableExists = metadata.listTables(session, targetTable.asQualifiedTablePrefix()).contains(targetTable);
```

Comment:
Pls consider a more economical alternative to listing internally all the tables in the schema.

Reply:
As discussed in the standup, we will be relying on the accessControl check which will throw TableNotFoundException if the table does not exist.

Reply:
relying on accessControl did not help here. Tes cases are failing.